src/reducedStream.ts

```diff
@@ -0,0 +1,133 @@
+   * startReducing does not always immediately emit a value since the value
+   * might have already been emitted when the state stream was created.
+   *
+   * initialState can be passed to reset the state of the stream to some other value.
+   * This can be useful for hot-reloading
+   */
+  startReducing = (
+    action$: ActionStream,
+    initialState = this.subject.value
+  ) => {
+    // Reset the subject value if it doesn't match the desired initialState
+    // This is normally used for hot-reloading
+    if (this.subject.value !== initialState) {
+      this.subject.next(initialState);
+    }
+    // NOTE: .subscribe(this.subject) does not work
+    // All tests pass with it, but in practice there are cases
+    // where the subject isn't updated as expected
+    this.subscription = action$
+      .pipe(
+        combineReducers(initialState, this.reducers, {
+          errorSubject: this.errorSubject,
+        }),
+        markName(this.name),
+        tag(this.name)
+      )
+      .subscribe({
+        complete: () => this.subject.complete(),
+        error: (err) => this.subject.error(err),
+        next: (state) => this.subject.next(state),
+      });
+    return this;
+  };
+
+  constructor(
+    name: string,
+    initialState: State,
+    reducers: RegisteredReducer<State, any>[],
+    errorSubject: Subject<any> = defaultErrorSubject
+  ) {
+    super();
+    this.name = name;
+    this.subject = new BehaviorSubject(initialState);
+    this.reducers = reducers;
+    this.errorSubject = errorSubject;
+  }
+
+  get state(): State {
+    return this.subject.getValue();
+  }
+
+  get closed() {
+    return this.subject.closed;
+  }
+
+  lift<R>(operator: Operator<State, R>): Observable<R> {
+    return this.subject.lift(operator);
+  }
+
+  unsubscribe() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+    this.subject.unsubscribe();
+  }
+
+  _trySubscribe(subscriber: Subscriber<State>): TeardownLogic {
+    return this.subject._trySubscribe(subscriber);
+  }
+
+  _subscribe(subscriber: Subscriber<State>): Subscription {
+    return this.subject._subscribe(subscriber);
+  }
+}
+
+/**
+ * Create a reduced state stream
+ *
+ * A reduced state stream is a stream that scans over an action stream and other
+ * stream to build up a state. It exposes it latest value like a Behaviour subject.
+ *
+ * To start reducing state, you must first call startReducing on the state stream.
+ *
+ * ```
+ * const myState$ = reducedStream(
+ *   'myState$',
+ *   initialState,
+ *   reducers
+ * );
+ *
+ * myState$.value === initialState // Will be true
+ *
+ * myState$.startReducing(action$) // To start reducing
+ * ```
+ *
+ * @param name The name of the stream, used for placing a marker and spy tag
+ * @param initialState The initial state of the stream
+ * @param reducers The reducers that build up the stream state
+ */
+export const reducedStream = <State>(
```

Comment:
If this is the main interface to create these kind of streams we should make it more clear what the purpose is. We cannot use these streams for e.g. view state streams, this is only meant for streams which should hold state over the whole life cycle of the application, view state streams on the other hand should start on subscription and stop when the according view is removed.

Reply:
Agree, but I would like to wait with making that clarifications until we have proper tooling for the view state streams. For now RxBeach only provides tooling for creating these kinds of "permanent" state streams and IMO `reducedStream` is a good name for that.

I can update the RxBeach docs with some examples and clarifications.

Reply:
I updated the PR and renamed `reducedStream` to `persistentReducedStream`